### PR TITLE
add flashBuy

### PR DESCRIPTION
**Pull request by ricklista**

### CODE REVIEW

File: contracts/FlashBuy.sol [hunk 1]
```diff
@@ -201,31 +81,39 @@ contract FlashBuy is IERC3156FlashBorrower, OwnableUpgradeable {
...
-            path,
-            address(this),
-            block.timestamp + 300
-        );
+        uint256 amountIn = IERC20(collateralReal).balanceOf(address(this)) - before;
+        IERC20(collateralReal).approve(address(dex), amountIn);
+
+        uint256 currentPrice = interaction.collateralPrice(collateral);
+        uint256 amountOut = (amountIn * currentPrice) / 1e18;
+        uint256 amountOutMin = (amountOut * (MAX_SLIPE - slip)) / MAX_SLIPE;
```
Comment (qingyang-lista) [inline contracts/FlashBuy.sol:108]: `amountOutMint` should bigger than minOut, otherwise throw an error

Comment (ricklista) [reply contracts/FlashBuy.sol:108]: image

dex will ensure amountOut >= amountOutMin

File: contracts/FlashBuy.sol [hunk 2]
```diff
@@ -149,31 +27,33 @@ contract FlashBuy is IERC3156FlashBorrower, OwnableUpgradeable {
     }
 
     IERC3156FlashLender public lender;
-    IAuctionProxy public auction;
-    IDEX dex;
+    IInteraction public interaction;
+    IDEX public dex;
+
+    uint256 constant public MAX_SLIPE = 10000;
```
Comment (oliver-lista) [inline contracts/FlashBuy.sol:33]: SLIPE is not a word. is this a typo ?

**Review by qingyang-lista — APPROVED** [review]
LGTM

**Pull request merged**